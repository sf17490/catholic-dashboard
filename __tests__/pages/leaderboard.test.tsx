import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

import Leaderboard from "@/pages/leaderboard";

describe("Leaderboard page", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(<Leaderboard />);
    expect(asFragment()).toMatchSnapshot();
  });
});
