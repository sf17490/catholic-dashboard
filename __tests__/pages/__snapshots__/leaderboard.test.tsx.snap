// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaderboard page matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <header
        class="govuk-header"
        data-testid="headerSection"
      >
        <div
          class="ml-10"
          data-testid="homePageLinkContainer"
        >
          <a
            class="govuk-header__link govuk-header__service-name"
            data-testid="homePageLink"
            href="/"
          >
            Catholic Data Dashboard
          </a>
        </div>
      </header>
    </div>
    <div
      class="m-10"
    >
      <h1
        class="govuk-heading-xl"
      >
         Leaderboard 
      </h1>
      <div
        data-testid="citation"
      >
        <b>
          Citation
        </b>
        <p
          data-testid="citationText"
        >
          The data on this page is calculated from the 2022 data in  
          <i>
            <a
              class="govuk-link"
              data-testid="citationLink"
              href="https://www.crs.org.uk/catholicism-in-numbers"
            >
              Catholicism in Numbers
            </a>
          </i>
          , a project by the Catholic Record Society.
        </p>
      </div>
      <div>
        <b>
          Understanding this table
        </b>
        <p>
          You can think of this table as measuring "Converts per capita". 
        </p>
        <p>
          For the purposes of this leaderboard, "practising Catholic" is defined as a person who fulfils their Sunday Mass obligation. 
        </p>
        <p>
          Each diocese is then ranked by the number of converts they are making per "practising" Catholic.
        </p>
        <b>
          Methodology
        </b>
        <p>
          For example, in 2022 in the Diocese of Hexham & Newcastle, 17,991 people attended Mass on a typical Sunday, and 146 people were received into the church that year. 17,991 divided by 146 is 123. It therefore "takes" 123 Catholics practising their faith in order to convert one person.
        </p>
        <p>
          In contrast, in the Diocese of Northampton, it takes 548 Catholics practising their faith to convert one person.
        </p>
      </div>
      <div
        class="bg-[var(--colour-grey-3)]"
        data-testid="dataDisplayBox"
      >
        <div
          class="m-6 pt-2 pb-0.25"
          data-testid="dataDisplayBoxContents"
        >
          <div>
            <div>
              <table
                class="govuk-table max-h-22"
                data-testid="table"
              >
                <caption
                  class="govuk-table__caption govuk-table__caption--m"
                >
                  Dioceses Ranked by Conversion Rate (2022)
                </caption>
                <thead
                  class="govuk-table__head"
                  data-testid="columnHeadings"
                >
                  <tr
                    class="govuk-table__row sticky-top-0"
                  >
                    <th
                      class="govuk-table__header"
                      data-testid="column0heading"
                      scope="col"
                    >
                      Rank
                    </th>
                    <th
                      class="govuk-table__header"
                      data-testid="column1heading"
                      scope="col"
                    >
                      Diocese
                    </th>
                    <th
                      class="govuk-table__header"
                      data-testid="column2heading"
                      scope="col"
                    >
                      Practising Catholics per Convert
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr
                    class="govuk-table__row"
                  >
                    <td
                      class="govuk-table__cell"
                    >
                      1
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      Hexham & Newcastle
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      123
                    </td>
                  </tr>
                  <tr
                    class="govuk-table__row"
                  >
                    <td
                      class="govuk-table__cell"
                    >
                      2
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      Birmingham
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      158
                    </td>
                  </tr>
                  <tr
                    class="govuk-table__row"
                  >
                    <td
                      class="govuk-table__cell"
                    >
                      3
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      Middlesbrough
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      174
                    </td>
                  </tr>
                  <tr
                    class="govuk-table__row"
                  >
                    <td
                      class="govuk-table__cell"
                    >
                      4
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      Liverpool
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      184
                    </td>
                  </tr>
                  <tr
                    class="govuk-table__row"
                  >
                    <td
                      class="govuk-table__cell"
                    >
                      5
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      Salford
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      187
                    </td>
                  </tr>
                  <tr
                    class="govuk-table__row"
                  >
                    <td
                      class="govuk-table__cell"
                    >
                      6
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      Cardiff
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      221
                    </td>
                  </tr>
                  <tr
                    class="govuk-table__row"
                  >
                    <td
                      class="govuk-table__cell"
                    >
                      7
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      Hallam
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      225
                    </td>
                  </tr>
                  <tr
                    class="govuk-table__row"
                  >
                    <td
                      class="govuk-table__cell"
                    >
                      8
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      Shrewsbury
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      232
                    </td>
                  </tr>
                  <tr
                    class="govuk-table__row"
                  >
                    <td
                      class="govuk-table__cell"
                    >
                      9
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      Plymouth
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      240
                    </td>
                  </tr>
                  <tr
                    class="govuk-table__row"
                  >
                    <td
                      class="govuk-table__cell"
                    >
                      10
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      Leeds
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      255
                    </td>
                  </tr>
                  <tr
                    class="govuk-table__row"
                  >
                    <td
                      class="govuk-table__cell"
                    >
                      11
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      Wrexham
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      264
                    </td>
                  </tr>
                  <tr
                    class="govuk-table__row"
                  >
                    <td
                      class="govuk-table__cell"
                    >
                      12
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      East Anglia
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      269
                    </td>
                  </tr>
                  <tr
                    class="govuk-table__row"
                  >
                    <td
                      class="govuk-table__cell"
                    >
                      13
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      Arundel & Brighton
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      274
                    </td>
                  </tr>
                  <tr
                    class="govuk-table__row"
                  >
                    <td
                      class="govuk-table__cell"
                    >
                      14
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      Brentwood
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      277
                    </td>
                  </tr>
                  <tr
                    class="govuk-table__row"
                  >
                    <td
                      class="govuk-table__cell"
                    >
                      15
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      Clifton
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      278
                    </td>
                  </tr>
                  <tr
                    class="govuk-table__row"
                  >
                    <td
                      class="govuk-table__cell"
                    >
                      16
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      Southwark
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      278
                    </td>
                  </tr>
                  <tr
                    class="govuk-table__row"
                  >
                    <td
                      class="govuk-table__cell"
                    >
                      17
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      Nottingham
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      280
                    </td>
                  </tr>
                  <tr
                    class="govuk-table__row"
                  >
                    <td
                      class="govuk-table__cell"
                    >
                      18
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      Lancaster
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      323
                    </td>
                  </tr>
                  <tr
                    class="govuk-table__row"
                  >
                    <td
                      class="govuk-table__cell"
                    >
                      19
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      Portsmouth
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      381
                    </td>
                  </tr>
                  <tr
                    class="govuk-table__row"
                  >
                    <td
                      class="govuk-table__cell"
                    >
                      20
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      Westminster
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      397
                    </td>
                  </tr>
                  <tr
                    class="govuk-table__row"
                  >
                    <td
                      class="govuk-table__cell"
                    >
                      21
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      Menevia
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      422
                    </td>
                  </tr>
                  <tr
                    class="govuk-table__row"
                  >
                    <td
                      class="govuk-table__cell"
                    >
                      22
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      Northampton
                    </td>
                    <td
                      class="govuk-table__cell"
                    >
                      548
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
      <div>
        <b>
          Limitations of this data
        </b>
        <p>
          This table should be interpreted with great caution. Please bear the following points in mind:
        </p>
        <ul
          class="govuk-list govuk-list--bullet"
        >
          <li>
            This table only considers data from the most recent year available (2022). It is possible that a diocese's ranking would change significantly if a different year was chosen.
          </li>
          <li>
            This ranking only considers receptions into the church - 
            <b>
              it does not include adult baptisms
            </b>
            . It is possible that a diocese's ranking would change significantly when converts from non-Christian backgrounds are included.
          </li>
        </ul>
        <p>
          Despite these limits we have published the ranking above in the belief that any measurement is better than no measurement at all.
        </p>
      </div>
    </div>
    <div
      class="govuk-footer"
      data-testid="footing"
    >
      <div
        class="ml-10"
        data-testid="footingInfoContainer"
      >
        <p
          data-testid="credit"
        >
          Inspired by the 
          <a
            class="govuk-footer__link"
            data-testid="UKHSAlink"
            href="https://ukhsa-dashboard.data.gov.uk/"
          >
            UKHSA Dashboard.
          </a>
           Styling reproduced under the 
          <a
            class="govuk-footer__link"
            data-testid="govLicenceLink"
            href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
          >
            Open Government Licence v3.0
          </a>
          . All our code is 
          <a
            class="govuk-footer__link"
            data-testid="dashboardGithubLink"
            href="https://github.com/sf17490/catholic-dashboard"
          >
            open-source
          </a>
          .
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
